features/step_definitions/registration_steps.rb

```diff
@@ -0,0 +1,15 @@
+Given(/^they are on the registration page$/) do
+  visit new_customer_registration_path
+end
+
+When(/^they fill in the form with valid details$/) do
+  fill_in "Username", with: "Finn"
+  fill_in "Email", with: "finn@finn.com"
+  fill_in "Password", with: "password"
+  fill_in "Confirm Password", with: "password"
+  click_button "Sign up"
+end
+
+Then(/^a Customer should be created$/) do
+  expect(Customer.count).to eq(1)
```

Comment:
I'm not sure this is adequate. If you add a check to see whether `expect(Customer.first.username).to eq('Finn')` I reckon you might not see anything... also so far I don't see any code that actually lets you sign in with a username, that would still be email by the looks of things. You need to subclass the sessions controller and change it's behaviour slightly.